src/ServiceClient/Implements/SocketClient/DeribitSocketClient.cs

```diff
@@ -0,0 +1,193 @@
+        StringBuilder stringBuilder = new();
+        byte[] readBuffer = new byte[readBufferSize];
+        var receiveValidStates = new[] { WebSocketState.Open, WebSocketState.CloseSent };
+        do
+        {
+            if (!receiveValidStates.Contains(webSocket.State)) return string.Empty;
+            response = await webSocket.ReceiveAsync(readBuffer, cancellationToken);
+            var result = Encoding.Default.GetString(readBuffer, 0, response.Count);
+            stringBuilder.Append(result);
+        }
+        while (!response.EndOfMessage);
+
+        return stringBuilder.ToString();
+    }
+
+    private async Task ConnectAsync(CancellationToken token)
+    {
+        if (webSocket.State == WebSocketState.Open) return;
+        await webSocket.ConnectAsync(new Uri(deribitOptions.WebSocketUrl), token);
+    }
+
+    private async Task<ActionResponse<T>?> ReadAsync<T>(CancellationToken token)
+        where T : class
+    {
+        var result = await ReadStringAsync(token);
+        return JsonSerializer.Deserialize<ActionResponse<T>>(result, jsonOptions);
+    }
+
+    public async Task DisconnectAsync(CancellationToken token)
+    {
+        await DisableHeartbeatAsync(token);
+        await webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, string.Empty, token);
+    }
+
+    public async Task CheckAvailabilityAsync(CancellationToken token)
+    {
+        if (webSocket.State != WebSocketState.Open)
+        {
+            await ConnectAsync(token);
+        }
+        await SendTestAsync(token);
+        await ReadAsync<object>(token);
+    }
+
+    public async Task AuthenticateAsync(CancellationToken token)
+    {
+        if (webSocket.State != WebSocketState.Open)
+        {
+            await ConnectAsync(token);
+        }
+        var data = new
+        {
+            grant_type = "client_credentials",
+            client_id = deribitOptions.ClientId,
+            client_secret = deribitOptions.ClientSecret,
+        };
+        await SendAsync("public/auth", data, token);
+        var credentials = await ReadAsync<AuthResult>(token);
+        Credentials = credentials?.Result;
+        await SetHeartbeatAsync(token);
+    }
+
+    public async Task SetHeartbeatAsync(CancellationToken token)
+    {
+        var data = new
+        {
+            interval = deribitOptions.HeartBeatInterval
+        };
+        await SendAsync("public/set_heartbeat", data, token);
+        await ReadAsync<object>(token);
+    }
+
+    public async Task DisableHeartbeatAsync(CancellationToken token)
+    {
+        await SendAsync("public/disable_heartbeat", new object(), token);
+        await ReadAsync<object>(token);
+    }
+
+    public async Task SubscribeAsync(CancellationToken token)
+    {
+        var data = new
+        {
+            channels = new[] { BookChannel, TickerChannel }
+        };
+
+        await SendAsync("private/subscribe", data, token);
+        await ReadAsync<object>(token);
```

Comment:
We're not using the result yet but here's the DTO for the response 

```
class SubscribeChannelsResponse : Response
{
    [JsonPropertyName("result")]
    public string[] Result { get; set; }
}
```

Reply:
I add this to another issue